core/templates/dev/head/components/top_navigation_bar/TopNavigationBarDirective.js

```diff
@@ -100,9 +101,19 @@ oppia.directive('topNavigationBar', [function() {
           $scope.$apply();
           // Close the sidebar, if necessary.
           SidebarStatusService.closeSidebar();
+          $scope.toggleText = '\uE5D2';
         });
 
-        $scope.toggleSidebar = SidebarStatusService.toggleSidebar;
+        $scope.toggleSidebar = (function() {
+          $scope.isSidebarShown = SidebarStatusService.isSidebarShown();
```

Comment:
Hi @himanshu-dixit -- deferring to @jaredsilver for the rest, but one quick comment: instead of adding all this logic, can you just add an ng-if="isSidebarShown()" to the HTML and put another `<i>` tag with ng-if="!isSidebarShown()"? Seems like it might be simpler.